s3prl/downstream/runner.py

```diff
@@ -432,41 +432,42 @@ def inference(self):
             self.downstream.model.inference(features, [filename])
 
     def push_to_huggingface_hub(self):
-        # Setup auth
-        hf_user = os.environ.get("HF_USERNAME")
-        hf_password = os.environ.get("HF_PASSWORD")
-        huggingface_token = HfApi().login(username=hf_user, password=hf_password)
-        HfFolder.save_token(huggingface_token)
-        print(f"HF Hub user: {hf_user}")
+        if self.args.hf_hub_org.lower() != "none":
+            organization = self.args.hf_hub_org
+        else:
+            organization = os.environ.get("HF_USERNAME")
+        huggingface_token = HfFolder.get_token()
+        print(f"[Runner] - Organisation to push fine-tuned model to: {organization}")
         
         # Create repo on the Hub
         upstream_model = self.args.upstream.replace("/", "__")
         repo_name = f"superb-s3prl-{upstream_model}-{self.args.downstream}-{str(uuid.uuid4())[:8]}"
         repo_url = HfApi().create_repo(
             token=huggingface_token,
             name=repo_name,
-            organization=hf_user,
+            organization=organization,
             exist_ok=True,
             private=True,
         )
-        print(f"Created Hub repo: {repo_url}")
+        print(f"[Runner] - Created Hub repo: {repo_url}")
 
         # Download repo and copy templates
-        REPO_PATH = self.args.expdir + "/hub_repo/"
+        HF_HUB_DIR = "hf_hub"
+        REPO_PATH = os.path.join(self.args.expdir, HF_HUB_DIR, repo_name)
         model_repo = Repository(
             local_dir=REPO_PATH, clone_from=repo_url, use_auth_token=huggingface_token
         )
         TEMPLATES_PATH = Path(f"./downstream/{self.args.downstream}/hf_hub_templates/")
         if TEMPLATES_PATH.exists():
             shutil.copytree(TEMPLATES_PATH, REPO_PATH, dirs_exist_ok=True)
         else:
-            print(f"No Hugging Face Hub template found for downstream task! Experiment files will still be pushed to the Hub in raw form")
+            print(f"[Runner] - No Hugging Face Hub template found for downstream task! Experiment files will still be pushed to the Hub in raw form")
 
         # Copy checkpoints, tensorboard logs, and args / configs
-        shutil.copytree(self.args.expdir, REPO_PATH, dirs_exist_ok=True, ignore=shutil.ignore_patterns("hub_repo"))
+        shutil.copytree(self.args.expdir, REPO_PATH, dirs_exist_ok=True, ignore=shutil.ignore_patterns(HF_HUB_DIR))
```

Comment:
This will copy everything in `expdir` to the Hub repo, including the config / tensorboard logs from previous runs. Presumably a good practice is to change `expdir` with each new experiment, so I'll leave this for now because disambiguating the files across runs is a bit tricky